src/index.ts

```diff
@@ -7,12 +7,80 @@ import * as dataRoot from "./data-root.js"
 import * as pathing from "./path.js"
 import * as ucan from "./ucan/internal.js"
 
-import { InitOptions, InitialisationError } from "./init/types.js"
-import { State, scenarioContinuation, scenarioNotAuthorised, validateSecrets } from "./auth/state.js"
+import { AppInitOptions, InitialisationError, LinkedAppInitOptions } from "./init/types.js"
+import { Permissions } from "./ucan/permissions.js"
+import { validateSecrets } from "./auth/state.js"
 import { bootstrapFileSystem, loadFileSystem } from "./filesystem.js"
 
 import FileSystem from "./fs/index.js"
 
+import { setImplementations } from "./setup.js"
+import { BASE_IMPLEMENTATION } from "./auth/implementation/base.js"
+import { USE_WNFS_IMPLEMENTATION } from "./auth/implementation/use-wnfs.js"
+import { LOBBY_IMPLEMENTATION } from "./auth/implementation/lobby.js"
+import { AppState } from "./auth/state/app.js"
+import { LinkedAppState } from "./auth/state/linkedApp.js"
+import * as appState from "./auth/state/app.js"
+import * as fissionAppState from "./auth/state/linkedApp.js"
+
+
+/**
+ * Check if we're authenticated and initiate the user's file system if 
+ * authenticated (can be disabled).
+ *
+ * See `loadFileSystem` if you want to load the user's file system yourself.
+ * NOTE: Only works on the main/ui thread, as it uses `window.location`.
+ *
+ */
+export async function app(options: AppInitOptions): Promise<AppState> {
+  options = options || {}
+
+  const { useWnfs = false } = options
+
+  if (useWnfs) {
+    setImplementations(USE_WNFS_IMPLEMENTATION)
+  } else {
+    setImplementations(BASE_IMPLEMENTATION)
```

Comment:
We could eventually make the base implementation the default because it assumes the least of any implementation.